error writing 1 byte into a region of size 0 when compiling with cmake

**Describe the bug**
Building with flags
-DWITH_GUI=ON
-DWITH_JAVA=OFF
-DWITH_TESTS=OFF
-DWITH_CSCORE=ON
-DWITH_WPILIB=ON
-DWITH_WPIMATH=ON
-DWITH_SHARED_LIBS=OFF
-DWITH_SIMULATION_MODULES=ON

causes the error 
```
simulation/halsim_ds_socket/src/main/native/cpp/DSCommPacket.cpp:244:36: error: writing 1 byte into a region of size 0 -Werror=stringop-overflow=]
  244 |     packet.descriptor.axisTypes[i] = data[1 + i];
```

**To Reproduce**
Steps to reproduce the behavior:
1. Generate cmake with the above mentioned flags
2. build project with cmake --build

**Expected behavior**
Libraries to be output with no errors

**Screenshots**
![scrn-2021-09-14-18-40-11](https://user-images.githubusercontent.com/35979547/133347861-3357d832-153b-4cb4-a57b-e2f49ed8ce07.png)


**Desktop (please complete the following information):**
 - OS: Void Linux  glibc x86_64
 - GCC version: 10.2.1

--- a/simulation/halsim_ds_socket/src/main/native/cpp/DSCommPacket.cpp
+++ b/simulation/halsim_ds_socket/src/main/native/cpp/DSCommPacket.cpp
@@ -239,7 +239,8 @@ void DSCommPacket::ReadJoystickDescriptionTag(wpi::span<const uint8_t> data) {
   packet.descriptor.name[nameLength] = '\0';
   int axesCount = data[0];
   packet.descriptor.axisCount = axesCount;
-  for (int i = 0; i < axesCount; i++) {
+  for (int i = 0;
+       i < std::min<int>(axesCount, sizeof(packet.descriptor.axisTypes)); i++) {
     packet.descriptor.axisTypes[i] = data[1 + i];
   }
   data = data.subspan(1 + axesCount);

--- a/simulation/halsim_ds_socket/src/main/native/cpp/DSCommPacket.cpp
+++ b/simulation/halsim_ds_socket/src/main/native/cpp/DSCommPacket.cpp
@@ -239,8 +239,9 @@ void DSCommPacket::ReadJoystickDescriptionTag(wpi::span<const uint8_t> data) {
   packet.descriptor.name[nameLength] = '\0';
   int axesCount = data[0];
   packet.descriptor.axisCount = axesCount;
-  for (int i = 0;
-       i < std::min<int>(axesCount, sizeof(packet.descriptor.axisTypes)); i++) {
+  for (int i = 0,
+           len = std::min<int>(axesCount, sizeof(packet.descriptor.axisTypes));
+       i < len; i++) {
     packet.descriptor.axisTypes[i] = data[1 + i];
   }
   data = data.subspan(1 + axesCount);

